packages/flutter/lib/src/semantics/semantics_service.dart

```diff
@@ -67,11 +70,21 @@ abstract final class SemanticsService {
     await SystemChannels.accessibility.send(event.toMap());
   }
 
+  /// Checks if announce is supported on the given platform.
+  bool isAnnounceSupported();
+}
+
+/// Default implementation of [SemanticsService].
+final class DefaultSemanticsService extends SemanticsService {
+  /// Creates a [DefaultSemanticsService].
+  const DefaultSemanticsService();
+
   /// Checks if announce is supported on the given platform.
   ///
   /// On Android the announce method is deprecated, therefore will return false.
   /// On other platforms, this will return true.
-  static bool isAnnounceSupported() {
+  @override
+  bool isAnnounceSupported() {
```

Comment:
@justinmc @chunhtai was talking with @jmagman and wanted to check out this pattern of approach. 'Announce' is deprecated on Android, so we want to got off of it on Android, but the liveRegion/announce support on various platforms is not consistent, so having this check would help determining the right code path.

Reply:
I think we should use AccessibilityFeatures to avoid having this kind of setting in random places